src/platform/silabs/wifi/wf200/WifiInterface.cpp

```diff
@@ -435,64 +458,67 @@ extern "C" sl_status_t sl_wfx_host_process_event(sl_wfx_generic_message_t * even
+    chip::ByteSpan scannedSsid(scan_result->ssid_def.ssid, scan_result->ssid_def.ssid_length);
+
+    // Verify that there was no requested SSID or that the SSID matches the requested SSID
+    VerifyOrReturn(requestedSsid.empty() || requestedSsid.data_equal(scannedSsid));
+
+    struct scan_result_holder * ap = reinterpret_cast<struct scan_result_holder *>(chip::Platform::MemoryAlloc(sizeof(*ap)));
+    VerifyOrReturn(ap != nullptr, ChipLogError(DeviceLayer, "Scan Callback: No Memory for scanned network."));
+
+    // Add Scan to the linked list
+    ap->next  = scan_save;
+    scan_save = ap;
+
+    // Copy scanned SSID to the output buffer
+    chip::MutableByteSpan outputSsid(ap->scan.ssid, WFX_MAX_SSID_LENGTH);
+    chip::CopySpanToMutableSpan(scannedSsid, outputSsid);
+    ap->scan.ssid_length = outputSsid.size();
+
+    // Set Network Security - We start by WPA3 to set the most secure type
+    ap->scan.security = WFX_SEC_UNSPECIFIED;
+    if (scan_result->security_mode.wpa3)
+    {
+        ap->scan.security = WFX_SEC_WPA3;
+    }
+    else if (scan_result->security_mode.wpa2)
+    {
+        ap->scan.security = WFX_SEC_WPA2;
+    }
+    else if (scan_result->security_mode.wpa)
     {
-        if (strcmp(scan_ssid, (char *) &scan_result->ssid_def.ssid[0]) != 0)
-            return;
+        ap->scan.security = WFX_SEC_WPA;
     }
-    if ((ap = (struct scan_result_holder *) (chip::Platform::MemoryAlloc(sizeof(*ap)))) == (struct scan_result_holder *) 0)
+    else if (scan_result->security_mode.wep)
     {
-        ChipLogError(DeviceLayer, "Scan: No Mem");
+        ap->scan.security = WFX_SEC_WEP;
     }
     else
     {
-        ap->next  = scan_save;
-        scan_save = ap;
-        /* Not checking if scan_result->ssid_length is < 33 */
-        chip::Platform::CopyString(ap->scan.ssid, sizeof(ap->scan.ssid), (char *) &scan_result->ssid_def.ssid[0]);
-        /* We do it in this order WPA3 first */
-        /* No EAP supported - Is this required */
-        ap->scan.security = WFX_SEC_UNSPECIFIED;
-        if (scan_result->security_mode.wpa3)
-        {
-            ap->scan.security = WFX_SEC_WPA3;
-        }
-        else if (scan_result->security_mode.wpa2)
-        {
-            ap->scan.security = WFX_SEC_WPA2;
-        }
-        else if (scan_result->security_mode.wpa)
-        {
-            ap->scan.security = WFX_SEC_WPA;
-        }
-        else if (scan_result->security_mode.wep)
-        {
-            ap->scan.security = WFX_SEC_WEP;
-        }
-        else
-        {
-            ap->scan.security = WFX_SEC_NONE;
-        }
-        ap->scan.chan = scan_result->channel;
-        ap->scan.rssi = scan_result->rcpi;
-        memcpy(&ap->scan.bssid[0], &scan_result->mac[0], kWifiMacAddressLength);
-        scan_count++;
+        ap->scan.security = WFX_SEC_NONE;
     }
+
+    ap->scan.chan = scan_result->channel;
+    ap->scan.rssi = (scan_result->rcpi - 220) / 2;
```

Comment:
why does the app need to convert to an RSSI value? where does this "formula" comes from

Reply:
No idea - this is what was done in another location. kept it as is since the rssi value with this formula matches the one when running with the 917.

When we refactor the WfiInterface file, this is going to be something to review.